l1-contracts/contracts/state-transition/chain-deps/facets/Admin.sol

```diff
@@ -119,6 +119,12 @@ contract AdminFacet is ZkSyncHyperchainBase, IAdmin {
         emit NewTransactionFilterer(oldTransactionFilterer, _transactionFilterer);
     }
 
+    function setBatchCommitDeadline(uint256 _newBatchCommitDeadline) external onlyAdmin {
```

Comment:
Why Admin should be able to change this parameter? I would say it is STM that decide

Reply:
Thanks, 

Trying to understand why this protection is enforced at the STM level.

My understand is that the `commit deadline` is a chain parameter. The value needs to consistent with the `CHAIN_STATE_KEEPER_BLOCK_COMMIT_DEADLINE_MS` parameter set by operator in chain config and in that case it NEEDS to be controlled by the chain admin.


Lets say chain operator decide to setup the `CHAIN_STATE_KEEPER_BLOCK_COMMIT_DEADLINE_MS` to 1 week. 
The value represents how much time the operator is willing to wait til it will submit a batch at "unprofitable" cost  (we can assume at chain operators lose money for each non-full batch it submit to ethereum)

In that case the commit deadline needs to be also set to 1week or more otherwise the chain may halt indefinitely.

So unless `CHAIN_STATE_KEEPER_BLOCK_COMMIT_DEADLINE_MS` is also a constant config for chain operators, I think we should let the chain operator set also its own `batch deadline`